Bottom: 65636beb13b8ae7b28da070da7c6de6af4e4ef9b
Top:    bff779db354e313c0743548a5ef661a763005024
Author: Sean Hefty <sean.hefty@intel.com>
Date:   2013-04-14 09:55:15 -0700

Refresh of rs-iwarp

---

diff --git a/src/rsocket.c b/src/rsocket.c
index 79aa53b..4caa93a 100644
--- a/src/rsocket.c
+++ b/src/rsocket.c
@@ -1542,16 +1542,16 @@ static int rs_post_msg(struct rsocket *rs, uint32_t msg)
 	wr.wr_id = rs_send_wr_id(msg);
 	wr.next = NULL;
 	if (!(rs->opts & RS_OPT_MSG_SEND)) {
-		wr.sg_list = &sge;
-		wr.num_sge = 1;
-		wr.opcode = IBV_WR_SEND;
-		wr.send_flags = IBV_SEND_INLINE;
-	} else {
 		wr.sg_list = NULL;
 		wr.num_sge = 0;
 		wr.opcode = IBV_WR_RDMA_WRITE_WITH_IMM;
 		wr.send_flags = 0;
 		wr.imm_data = htonl(msg);
+	} else {
+		wr.sg_list = &sge;
+		wr.num_sge = 1;
+		wr.opcode = IBV_WR_SEND;
+		wr.send_flags = IBV_SEND_INLINE;
 	}
 
 	return rdma_seterrno(ibv_post_send(rs->cm_id->qp, &wr, &bad));
